gatsby-node.ts

```diff
@@ -54,5 +57,93 @@ exports.createPages = async ({ graphql, actions }: CreatePagesArgs) => {
     }),
   );
 
+  await handleFrontPageData(graphql, createPage);
+
   return Promise.all(promises);
 };
+
+async function handleFrontPageData(graphql: CreatePagesArgs['graphql'], createPage: Actions['createPage']) {
+  const frontPageResponse = await makeRequest(graphql, getFrontPage);
+  const frontPage = (frontPageResponse.data as { strapiFrontPage: StrapiFrontPage }).strapiFrontPage as StrapiFrontPage;
+
+  for (const navigationItem of frontPage.navigation || []) {
+    // If the page's id is null, the page is not published so let's skip it
+    if (!navigationItem?.page?.id) continue;
+
+    if (navigationItem) await createNavigationLevel(graphql, createPage, navigationItem);
+  }
+}
+
+async function createNavigationLevel(
+  graphql: CreatePagesArgs['graphql'],
+  createPage: Actions['createPage'],
+  data: StrapiFrontPageNavigation,
+): Promise<void> {
+  // If the page's id is null, the page is not published so let's skip it
+  if (!data.id) return;
```

Comment:
Minor notice: I think this can be removed since check is already done in caller function handleFrontPageData

Reply:
By keeping this here, typescript knows that `data.id` is defined and I don't have to assure it later :)